### PR TITLE
Exception -> warning for unsupported backend opt

### LINKED ISSUE #834: Verbose option not supported by numpy backend
For teaching purposes, we often use the verbose=True option when compiling a stencil. This option is not supported by the numpy backend. It is very annoying when switching back and forth between different backends in exemplary gt4py code. I would suggest to implement a verbose option into the numpy backend, even if it does absolutely nothing.

### PR DESCRIPTION
**Pull request by havogt**
Fixes #834

### CODE REVIEW

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
bors try

File: tests/test_integration/test_stencil_object.py [hunk 1]
```diff
@@ -64,3 +65,12 @@ def runit(*args, **kwargs) -> float:
...
     assert cleaned_cache_time > fast_time
+
+
+@pytest.mark.parametrize("backend", ALL_BACKENDS)
+@pytest.mark.parametrize("backend_opts", [{"verbose": True}])
+def test_common_stencil_options(backend, backend_opts):
+    @gtscript.stencil(backend=backend, **backend_opts)
+    def foo(f: Field[float]):
+        with computation(PARALLEL), interval(...):  # type: ignore
+            f = 42.0  # noqa F841
```
Comment (jdahm) [inline tests/test_integration/test_stencil_object.py:76]: Is this testing that the `verbose` option is allowed in all backends, even if it is not defined as an option in some? Can you capture the warning and assert that it happens?

Comment (havogt) [reply tests/test_integration/test_stencil_object.py]: I changed the test to actually test to use a random option that none of the backends supports and check for the warning. I think it documents better the purpose of this change.

**Comment by havogt** [comment]
bors try

**Review by jdahm — APPROVED** [review]
Looks great!

**Comment by havogt** [comment]
Thanks for reviewing.

**Pull request merged**